internal/manifests/collector/adapters/config_to_ports.go

```diff
@@ -44,7 +43,7 @@ func (c ComponentType) String() string {
 }
 
 // ConfigToComponentPorts converts the incoming configuration object into a set of service ports required by the exporters.
-func ConfigToComponentPorts(logger logr.Logger, cType ComponentType, config map[interface{}]interface{}) ([]v1beta1.PortsSpec, error) {
+func ConfigToComponentPorts(logger logr.Logger, cType ComponentType, config map[interface{}]interface{}) ([]corev1.ServicePort, error) {
```

Comment:
I needed to change this back to `corev1.ServicePort` to avoid a circular dependency. 

Originally it was using the type from `corev1` but it was changed here https://github.com/open-telemetry/opentelemetry-operator/pull/2763/files#diff-39548fb5f0030535966b3fb437c08814c648216539e297c034f3da6ac5b43a9bR47

Reply:
doing this is going to make it _very_ hard to complete #2603 because we want to be able to use those types. Is it possible to avoid this by moving the methods?